src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/LicenceService.kt

```diff
@@ -31,18 +27,16 @@ class LicenceService(
     val curfew = licenceData.curfew
     val proposedAddress = licenceData.proposedAddress
 
-    return curfew?.curfewHours?.let { formatCurfewHoursObject(curfew.curfewHours) }?.let {
-      HdcLicence(
-        licenceId = licence.id,
-        curfewAddress = getAddress(curfew, cas2Referral, proposedAddress),
-        firstNightCurfewHours = curfew.firstNight,
-        // curfewHours referred to as curfewTimes in CVL as going forward a more suitable name and had to distinguish between the two different curfew data formats
-        curfewTimes = it,
-      )
-    }
+    return HdcLicence(
+      licenceId = licence.id,
+      curfewAddress = getAddress(curfew, cas2Referral, proposedAddress),
+      firstNightCurfewHours = curfew?.firstNight?.let { transformToModelFirstNight(it) },
+      // curfewHours referred to as curfewTimes in CVL as going forward a more suitable name and had to distinguish between the two different curfew data formats
+      curfewTimes = transformToModelCurfewTimes(curfew?.curfewHours),
```

Comment:
I was going to make a comment about whether this should be: 
`curfew?.let { transformToModelCurfewTimes(curfew.curfewHours) }`
But actually I'm not sure it should be, because 1) it wouldn't affect the code in any other way, as null still has to be handled with `curfewHours` being nullable and 2) this way we don't have to handle `curfewTimes` being nullable in the UI. Not sure if arguably we _should_ "have" to handle it though, maybe one to check with Andy again.